tests/sentry/api/endpoints/test_project_key_stats.py

```diff
@@ -1,45 +1,101 @@
-from sentry import tsdb
+from sentry.constants import DataCategory
 from sentry.models import ProjectKey
 from sentry.testutils import APITestCase
+from sentry.testutils.cases import OutcomesSnubaTest, SnubaTestCase
+from sentry.testutils.helpers.datetime import before_now
+from sentry.utils.outcomes import Outcome
 
 
-class ProjectKeyStatsTest(APITestCase):
+class ProjectKeyStatsTest(OutcomesSnubaTest, SnubaTestCase, APITestCase):
     def setUp(self):
+        super().setUp()
         self.project = self.create_project()
         self.key = ProjectKey.objects.create(project=self.project)
         self.login_as(user=self.user)
         self.path = f"/api/0/projects/{self.project.organization.slug}/{self.project.slug}/keys/{self.key.public_key}/stats/"
 
     def test_simple(self):
-        tsdb.incr(tsdb.models.key_total_received, self.key.id, count=3)
-        tsdb.incr(tsdb.models.key_total_blacklisted, self.key.id, count=1)
-
+        self.store_outcomes(
```

Comment:
If we're not testing the snuba query here, could we avoid the external dependency by mocking `massage_outcomes_result`?

That being said, having one test rely on snuba would alert us if any snuba changes break this endpoint.

Reply:
This test does interact with snuba though. The `store_outcomes` method stores outcomes in snuba and the endpoint queries those records.